src/term/transform/linearize.rs

```diff
@@ -72,6 +72,35 @@ fn term_to_affine(
         }
       }
     }
+    Term::Let { pat: LetPat::Tup(l_nam, r_nam), val, nxt } => {
+      let fst_uses = l_nam.as_ref().map(|l_nam| *var_uses.get(l_nam).unwrap()).unwrap_or(0);
+      let snd_uses = r_nam.as_ref().map(|r_nam| *var_uses.get(r_nam).unwrap()).unwrap_or(0);
+
+      match (fst_uses, snd_uses) {
+        (0, 0) => *nxt,
+        (_, 0) => match *val {
+          Term::Tup { fst, .. } => {
+            let let_ = Term::Let { pat: LetPat::Var(l_nam.unwrap()), val: fst, nxt };
+            term_to_affine(let_, var_uses, let_bodies)?
+          }
+          _ => unimplemented!(),
```

Comment:
Tup is not the only body that we expect here. For example:
`((@x let (a, b) = x; (+ a b)) (1, 2))`
This is a valid term that in the context of this pass should remain the same.